projects/core/src/occ/adapters/product/occ-product-search.adapter.ts

```diff
@@ -22,12 +22,14 @@ import { ConverterService } from '../../../util/converter.service';
 import { Occ } from '../../occ-models/occ.models';
 import { OccEndpointsService } from '../../services/occ-endpoints.service';
 import { OCC_HTTP_TOKEN } from '../../utils';
+import { Router } from '@angular/router';
 @Injectable()
 export class OccProductSearchAdapter implements ProductSearchAdapter {
   constructor(
     protected http: HttpClient,
     protected occEndpoints: OccEndpointsService,
-    protected converter: ConverterService
+    protected converter: ConverterService,
+    @Optional() protected router: Router
```

Comment:
Any change of constructor is considered as a breaking change. For the changes in constructor, to avoid breaking changes, we are using constructor overloading. Explained [here](https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions).

But there is no need in this case to add router to constructor.
You can just declare service as class property using `inject()`

```
protected router = inject(Router, {
    optional: true,
});
```